src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc

```diff
@@ -4,5 +4,319 @@ title: Integrating ElevenLabs into Appwrite
+```
+
+Next add code to validate the body of the request and initialise the Appwrite SDK:
+```js
+const endpoint = process.env.APPWRITE_ENDPOINT || "https://cloud.appwrite.io/v1"
+
+const client = new Client()
+  .setEndpoint()
+  .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
+  .setKey(process.env.APPWRITE_API_KEY);
+
+try {
+  throwIfMissing(req.body, ["text"]);
+} catch (err) {
+  return res.json({ ok: false, error: err.message }, 400);
+}
+```
+
+Finally, Make a request to the ElevenLabs API and return the response:
+
+```js
+let body = {
+  accent: req.body.accent || "british",
+  accent_strength: 1.0,
+  age: req.body.age || "young",
+  gender: req.body.gender || "female",
+  text: req.body.text,
+};
+
+const resp = await fetch(
+  "https://api.elevenlabs.io/v1/voice-generation/generate-voice",
+  {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "xi-api-key": process.env.ELEVENLABS_API_KEY,
+    },
+    body: JSON.stringify(body),
+  },
+);
+
+if (resp.status !== 200) {
+  error(await resp.text());
+  return res.json({ ok: false, error: "Failed to generate audio" }, 500);
+}
+```
+
+This code will send the prompt to the ElevenLabs API and return the audio as a blob,
+additionally it'll also catch any errors we could encounter and reports them for easy debugging.
+{% /section %}
+
+{% section #step-7 step=7 title="Store Audio in Appwrite Storage" %}
+
+Now you can store the audio file in Appwrite Storage for easy retrieval later:
+
+```js
+const storage = new Storage(client);
+
+let file = await storage.createFile(
+  process.env.APPWRITE_BUCKET_ID,
+  ID.unique(),
+  InputFile.fromBlob(await resp.blob(), "audio.mp3"),
+  [Permission.read(Role.any())],
+);
+```
+
+To show it to the user, parse the download URL from Appwrite and return it in the response:
+
+```js
+const url = `${endpoint}/storage/buckets/${process.env.APPWRITE_BUCKET_ID}/files/${file.$id}/view?project=${process.env.APPWRITE_FUNCTION_PROJECT_ID}`;
+
+return res.json({ ok: true, response: url });
+```
+
+This should finish up the function, Deploy it to Appwrite by pushing to the git repository created earlier.
+{% /section %}
+
+{% section #step-8 step=8 title="Test the function" %}
+Now that the function is deployed, test it by visiting the function URL in your browser.
+This should show the UI created earlier and to test it, write a prompt and click the submit button. After a brief moment you should see the audio appear below the input.
+
+![Testing the function](/images/docs/ai/integrations/elevenlabs/demo.png)
+{% /section %}
+
+{% section #step-9 step=9 title="Next steps" %}
+In this tutorial, you've created a simple web page that allows you to interact with the ElevenLabs API using Appwrite Functions. You've learned how to create a function that interacts with an external API, upload audio to storage, how to deploy it to Appwrite, and how to create an interactive web page that uses the function.
+{% /section %}
```

Comment:
This section is not required. We aren't really giving any next steps here. Lets skip it.